hazelcast/src/main/java/com/hazelcast/internal/diagnostics/InvocationProfilerPlugin.java

```diff
@@ -0,0 +1,86 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.internal.diagnostics;
+
+import com.hazelcast.internal.util.LatencyDistribution;
+import com.hazelcast.spi.impl.NodeEngineImpl;
+import com.hazelcast.spi.impl.operationservice.impl.InvocationRegistry;
+import com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl;
+import com.hazelcast.spi.properties.HazelcastProperty;
+
+import java.util.Map;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+/**
+ * A {@link DiagnosticsPlugin} that displays invocation latency information.
+ */
+public class InvocationProfilerPlugin extends DiagnosticsPlugin {
+
+    public static final HazelcastProperty PERIOD_SECONDS = new HazelcastProperty(
+            "hazelcast.diagnostics.invocation-profiler.period.seconds", 5, SECONDS);
+
+    private final InvocationRegistry invocationRegistry;
+    private final long periodMs;
+
+    public InvocationProfilerPlugin(NodeEngineImpl nodeEngine) {
+        super(nodeEngine.getLogger(PendingInvocationsPlugin.class));
+        OperationServiceImpl operationService = nodeEngine.getOperationService();
+        this.invocationRegistry = operationService.getInvocationRegistry();
+        this.periodMs = nodeEngine.getProperties().getMillis(PERIOD_SECONDS);
+    }
+
+    @Override
+    public long getPeriodMillis() {
+        return periodMs;
+    }
+
+    @Override
+    public void onStart() {
+        logger.info("Plugin:active: period-millis:" + periodMs);
+    }
+
+    @Override
+    public void run(DiagnosticsLogWriter writer) {
+        writer.startSection("InvocationProfiler");
+
+        for (Map.Entry<Class, LatencyDistribution> entry : invocationRegistry.latencyDistributions().entrySet()) {
+            LatencyDistribution distribution = entry.getValue();
+            if (distribution.count() == 0) {
+                continue;
+            }
+
+            writer.startSection(entry.getKey().getName());
+            writer.writeKeyValueEntry("count", distribution.count());
+            writer.writeKeyValueEntry("totalTime(us)", distribution.totalMicros());
+            writer.writeKeyValueEntry("avg(us)", distribution.avgMicros());
+            writer.writeKeyValueEntry("max(us)", distribution.maxMicros());
+
+            writer.startSection("latency-distribution");
+            for (int bucket = 0; bucket < distribution.bucketCount(); bucket++) {
+                long value = distribution.bucket(bucket);
+                if (value > 0) {
+                    writer.writeKeyValueEntry(StoreLatencyPlugin.LATENCY_KEYS[bucket], value);
```

Comment:
Should be `LatencyDistribution.LATENCY_KEYS[bucket]`

Reply:
Fixed